action/protocol/staking/handlers.go

```diff
@@ -371,15 +389,19 @@ func (p *Protocol) handleTransferStake(ctx context.Context, act *action.Transfer
 	}
 
 	newOwner := act.VoterAddress()
-	bucket, fetchErr := p.fetchBucketAndValidate(csm, actionCtx.Caller, act.BucketIndex(), true, false)
+	bucket, fetchErr := p.fetchBucket(csm, act.BucketIndex())
+	if fetchErr != nil {
+		return log, fetchErr
+	}
+	fetchErr = p.validateTransferStakeBucket(featureCtx, csm, bucket, actionCtx.Caller)
 	if fetchErr != nil {
 		if featureCtx.ReturnFetchError ||
 			fetchErr.ReceiptStatus() != uint64(iotextypes.ReceiptStatus_ErrUnauthorizedOperator) {
 			return log, fetchErr
 		}
 
 		// check whether the payload contains a valid consignment transfer
-		if consignment, ok := p.handleConsignmentTransfer(csm, actionCtx, act, bucket); ok {
+		if consignment, ok := p.handleConsignmentTransfer(csm, ctx, act, bucket); ok {
```

Comment:
why need to change this?

Reply:
b/c `ContainsSelfStakingBucket` is called in the func